### ISSUE TITLE
Research about smart contracts (is it something we need)

### ISSUE THREAD

**Issue by ghanem025** [ghanem025/Blockchain-app#19]
We have not looked at smart contracts for our blockchain (might not have to necessarily). However I have been looking at some tools like vyper that may help us create a smart contract for our blockchain. this can make the data more secure, strengthen the anonymity of our blockchain and reduce the risk 

some points from an article that talks about the benefits of smart contracts (https://corporatefinanceinstitute.com/resources/valuation/smart-contracts/)
```
Blockchain can store the encoded health records of patients with a private key. 
Only specific individuals would be granted access to the records for privacy concerns. 
Similarly, research can be conducted confidentially and securely using smart contracts.

All hospital receipts of patients can be stored on the blockchain and automatically shared with insurance companies as proof of service. 
Moreover, the ledger can be used for different activities, such as managing supplies, supervising drugs, and [regulation compliance]
(https://www.fda.gov/drugs).
```
TL:DR smart contracts bump this project to a new level, but it will not be easy. it is however worth looking into.
